src/noisy_graph_states/libs/patterns.py

```diff
@@ -0,0 +1,231 @@
+# -*- coding: utf-8 -*-
+"""
+A collection of useful functions to translate between measurement patterns and strategies.
+"""
+
+import graphepp as gg
+from itertools import permutations
+from collections import defaultdict
+import noisy_graph_states.libs.graph as gt
+
+
+# Before using the function in this file. We define the concepts of sequence and pattern.
+# sequence: ordered list of tuples ("z", index), ("y", index) or ("x", index, b0) where b0 is optional and automatically
+# determined if not given.
+# pattern: list of strings with "x" "y" "z" or "." (no measurement)
+
+
+def _default_projector_matching():
+    return {"x": "x", "y": "y", "z": "z"}
+
+
+def sequence_to_pattern(sequence: list[tuple], graph: gg.Graph):
+    """Find measurement pattern corresponding to graph measurements.
+
+    Every sequence of measurements in the graph state picture (i.e.,
+    with explicitly undoing the byproduct operators at every step)
+    corresponds to a measurement pattern of Pauli measurements
+    that postpones all corrections operations to the very end.
+    There are multiple sequences that lead to the same
+    pattern.
+
+    This is explained in the appendix of
+    > Mor-Ruiz, Maria Flors, Julius Wallnöfer, and Wolfgang Dür.
+    > "Imperfect quantum networks with tailored resource states."
+    > arXiv preprint arXiv:2403.19778 (2024).
```

Comment:
```suggestion
    | This is explained in the appendix of
    | Maria Flors Mor-Ruiz, , Julius Wallnöfer, and Wolfgang Dür.
    | "Imperfect quantum networks with tailored resource states."
    | arXiv preprint arXiv:2403.19778 (2024).
```

This formatting will make it look nice and on separate lines in the docs.

Also should all the names be in the same first name - last name order?